lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua

```diff
@@ -258,6 +258,172 @@ function EGP:DrawLine( x, y, x2, y2, size )
 	end
 end
 
+function EGP:DrawPath( vertices, size, closed )
+	if size < 1 then size = 1 end
+	local num = #vertices
+
+	if size == 1 then
+		if closed then
+			vertices[num+1] = vertices[1]
+			num = num+1
+		end
+		local last = vertices[1]
+		for i=2, num do
+			local v = vertices[i]
+			surface.DrawLine( last.x, last.y, v.x, v.y )
+			last = v
+		end
+	else
+		if closed then
+			-- pad array to simplify calculation
+			vertices[num+1] = vertices[1]
+			vertices[num+2] = vertices[2]
+			num = num+1
+		end
+
+		local x1 = vertices[1].x
+		local y1 = vertices[1].y
+
+		for i=2, num do
+			local x2 = vertices[i].x
+			local y2 = vertices[i].y
+
+			local tempx = x2
+			local tempy = y2
+
+			local len = math.sqrt( (x2-x1) ^ 2 + (y2-y1) ^ 2 )
+			local dir = {x=(x2-x1)/len, y=(y2-y1)/len}
+
+			if closed and i==2 then
+				-- Offset start point to match with closing line
+				local x0 = vertices[num-1].x
+				local y0 = vertices[num-1].y
+				local len0 = math.sqrt( (x1-x0) ^ 2 + (y1-y0) ^ 2 )
+				local dir0 = {x=(x1-x0)/len0, y=(y1-y0)/len0}
+
+				local ang = math.abs(math.acos(dir0.x*dir.x + dir0.y*dir.y)) -- angle between previous line segment and this one
+				if math.deg(ang)<=90 then
```

Comment:
local dot = dir0.x*dir.x + dir0.y*dir.y
if dot>=0 then
local shift = size/2 * math.tan(math.acos(dot)/2)

Should work quicker, especially if ang>90

Reply:
Oh wait there's a better way.

Reply:
Try the first one first and if it works then
```lua
local dot = dir0.x*dir.x + dir0.y*dir.y
if dot>=0 then
local halftanang = math.sqrt(((dir0.x-dir.x)^2 + (dir0.y-dir.y)^2)/((dir0.x+dir.x)^2 + (dir0.y+dir.y)^2))
local shift = size/2 * halftanang
```

Reply:
I'll run a benchmark between these two actually and see which is better.

Reply:
They are the same, so probably better to stick with the math.tan(math.acos(dot)/2) version